### PR TITLE
fix(module: table): fix collection modified exception

### PR DESCRIPTION
**Pull request by diegofrata**
[[中文版模板 / Chinese template](https://github.com/ant-design-blazor/ant-design-blazor/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [X] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [X] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

This change does a few things:

1) Fix the issue ``System.InvalidOperationException: Collection was modified; enumeration operation may not execute.`` reported by an user in the Slack channel. 

2) Removes a potenticial performance issue by avoiding multiple evaluations of a same IEnumerable when possible;

3) Introduces a way of recording the markup generated by a component at the end of a test so it can be compared against future changes.

The last item is a proposition on how we can compare results of tests without a ridiculous amount of effort -- all you need to do is run the test once to record the output and the next executions will be measured against it. Would be interested in hearing thouhgts about it.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: tests/table/TableTests.cs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+using Bunit;
+using Xunit;
+
+namespace AntDesign.Tests.Table
+{
+    public class TableTests : AntDesignTestBase
+    {
+        class Person
+        {
+<<<<<<< Updated upstream
```
Comment (ElderJames) [inline tests/table/TableTests.cs:12]: Is there some conflict that hasn't been resolved?

**Review by ElderJames — CHANGES_REQUESTED** [review]
Thanks for your contribution. Please resolve the git conflicts.

File: tests/table/TableTests.cs (hunk 1 above)
Comment (diegofrata) [reply tests/table/TableTests.cs]: I think I did something weird last night when trying to push. 😂 It's fixed now.

**Comment by diegofrata** [comment]
@ElderJames the PR should be good now.

**Review by ElderJames — APPROVED** [review]
Thank you.

**Pull request merged**